grabbit/core.py

```diff
@@ -185,9 +188,19 @@ def index(self):
         # Loop over all files
         for root, directories, filenames in os.walk(self.root, topdown=True):
 
-            # Exclude directories from further search if they match exclude regex
-            if self.exclude_dir is not None:
-                directories[:] = [d for d in directories if not re.match(self.exclude_dir, d)]
+            # Exclude/include top-level directories from further search
+            #if they match exclude regex
+            if root == self.root:
+                if self.index_regex:
```

Comment:
I don't see any reason why this should apply only to directories... probably all files should be screened against include and exclude. I'd probably move this into `_is_valid` (probably after the PR for validation is merged, to prevent conflicts). I.e., after each file is checked for validity in the base BIDS spec, we can then inspect it against the regexes (if needed) to see if it passes.

Reply:
So do you mean the `_validate_file` function? So if you want to only exclude a directory you can a regex with a "/"